client/src/js/models/SessionModel.js

```diff
@@ -0,0 +1,92 @@
+/* global gapi */
+
+var _ = require('underscore');
+var Backbone = require('backbone');
+var UserModel = require('./UserModel.js');
+
+var SessionModel = Backbone.Model.extend({
+  defaults: {
+    logged: false,
+    userId: ''
+  },
+
+  initialize: function() {
+    this.user = new UserModel({});
+  },
+
+  loginSessionUser: function(googleUser) {
+    var profile = googleUser.getBasicProfile();
+
+    // Get attributes from user's gmail account and assign to
+    // the current user in this session. 
+    this.updateSessionUser({
+      id: googleUser.getAuthResponse().id_token,
+      name: profile.getName(),
+      email: profile.getEmail(),
+      image: profile.getImageUrl()
+    });
+
+    // Set the SessionModel's values and show that a user is
+    // now logged in.
+    this.set({ userId: this.user.get('id'), logged: true });
+  },
+
+  logoutSessionUser: function() {
+
+    // Clear and resets GoogleUser attributes to UserModel
+    // default values.
+    this.updateSessionUser(this.user.defaults);
+
+    // Reset the SessionModel's values to defaults and show 
+    // that a user is now logged out.
+    this.set({ userId: this.user.get('id'), logged: false });
+  },
+
+  updateSessionUser: function(userData) {
+
+    // A user model is passed into this function.
```

Comment:
Not quite, the user model itself is not passed, just the data that we received from the authentication to be saved in the user model.